asab/library/providers/filesystem.py

```
@@ -57,113 +74,196 @@ def __init__(self, library, path, layer, *, set_ready=True):
-			return None
 
-		except IsADirectoryError:
+	async def read(self, path: str) -> typing.Optional[typing.IO]:
+		# Tenant first
+		try:
+			tenant_node_path = self.build_path(path, tenant_specific=True)
+			if os.path.isfile(tenant_node_path):
+				return io.FileIO(tenant_node_path, 'rb')
+		except Exception:
+			# keep behavior similar to zk: just fall back
+			pass
```

Ruff (0.14.8)

`try`-`except`-`pass` detected, consider logging the exception

(S110)